diff --git a/configure.ac b/configure.ac
index 626d024..c1eb141 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1235,7 +1235,7 @@ if test $ol_with_tls = openssl || test $ol_with_tls = auto ; then
 
 		AC_CHECK_LIB(ssl, SSL_export_keying_material_early,
 			[have_openssl=yes], [have_openssl=no],
-			[-lcrypto])
+			[-lcrypto -lpthread])
 
 		if test $have_openssl = yes ; then
 			ol_with_tls=openssl
